torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1977,91 +2058,34 @@ void ProcessGroupNCCL::watchdogHandler() {
       if (!terminateProcessGroup_.load()) {
         work.checkAndSetException();
       }
+      // Then check if work has timed out
+      bool timedout = work.checkTimeout();
+
+      // Report desync state in case of timeout (if TORCH_NCCL_DESYNC_DEBUG is
+      // turned on; otherwise, run() is no-op)
+      if (timedout) {
+        desyncDebugger_.run();
+      }
+
+      // If work hits an exception (either an error or timeout)
```

Comment:
This would make it harder to debug whether the execption is nccl error or timeout, We'd better clearly differentiate the two types of logs, because usually there are triggered by very different reasons.

Reply:
This is old code. I didn't change the behavior. Can we address in a new PR if needed?